.travis.yml

```diff
@@ -1,26 +1,71 @@
-language: cpp
-
-services:
-  - docker
-
-cache:
-  ccache: true
-
 env:
   global:
     - CCACHE_DIR=$HOME/.ccache
     - PX4_DOCKER_REPO=px4io/px4-dev-ros:2018-04-22
-    - BUILD_WITH_CMAKE = "mkdir Build; cd Build; cmake ..; make"
-    - BUILD_WITH_CATKIN = "source /opt/ros/kinetic/setup.bash; mkdir -p ~/catkin_ws/src; cd ~/catkin_ws/src; catkin_init_workspace; ln -s ${TRAVIS_BUILD_DIR} .; cd ~/catkin_ws; catkin_make"
-    - VALIDATE_SDF_SCHEMAS = "source ./scripts/validate_sdf.bash"
 
-  matrix:
-    # Cmake build
-    - STEP=${BUILD_WITH_CMAKE}
-    # Catkin build
-    - STEP=${BUILD_WITH_CATKIN}
-    # Validate SDF schemas
-    - STEP=${VALIDATE_SDF_SCHEMAS}
+matrix:
+    include:
+        - os: linux
+          language: cpp
+          services:
+            - docker
+          cache:
+            ccache: true
+          env: # Cmake build
+            - BUILD="mkdir Build; cd Build; cmake ..; make -j$(nproc) -l$(nproc)"
+        - os: linux
+          language: cpp
+          services:
+            - docker
+          cache:
+            ccache: true
+          env: # Catkin build
+            - BUILD="source /opt/ros/kinetic/setup.bash; mkdir -p ~/catkin_ws/src; cd ~/catkin_ws/src; catkin_init_workspace; ln -s ${TRAVIS_BUILD_DIR} .; cd ~/catkin_ws; catkin_make -j$(nproc) -l$(nproc)"
+        - os: linux
+          language: cpp
+          services:
+            - docker
+          cache:
+            ccache: true
+          env: # Validate SDF schemas
+            - BUILD="source ./scripts/validate_sdf.bash"
+        - os: osx
+          osx_image: xcode9.3
+          language: cpp
+          cache:
+            ccache: true
+          env:
+            - GAZEBO_VERSION="gazebo8"
+        - os: osx
+          osx_image: xcode9.3
+          language: cpp
+          cache:
+            ccache: true
+          env:
+            - GAZEBO_VERSION="gazebo9"
+
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+    brew update;
+    git clone --depth 1 https://github.com/mavlink/c_library_v2.git /usr/local/include/mavlink/v2.0;
+    rm -rf /usr/local/include/mavlink/v2.0/.git;
+    fi
+
+install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+    sudo easy_install pip;
+    sudo -H pip install -I nose tornado rospkg Jinja2 -t /Users/travis/Library/Python/2.7/bin;
+    brew install eigen graphviz protobuf opencv3;
+    brew upgrade automake cmake libxml2;
+    brew link --overwrite --dry-run python@2 numpy;
```

Comment:
When you need to use force or overwrite flags you already know the toolchain is messed up 😈